File: Algorithms/Sorting/Insertion_Sort-Part_2/main.py
### Insertion Sort - Part 2 - Solution

def insertionSort2(nums):
    for i in range(1, len(nums)):
        comp, prev = nums[i], i-1
        while (prev >= 0) and (nums[prev] > comp):
            nums[prev+1] = nums[prev]
            prev -= 1
        nums[prev+1] = comp
        print(*nums) 

n = int(input())
nums = list(map(int, input().split()[:n]))
insertionSort2(nums)

File: Algorithms/Sorting/Insertion_Sort-Part_2/main.cpp
// Insertion Sort - Part 2 - Solution

#include <iostream>

void printVector(const int n, const int nums[]) {
    for (int i{}; i < n; i++)
        std::cout << nums[i] << " ";
    std::cout << '\n';
}

void insertionSort2(const int n, int nums[]) {
    int cmp, prev;
    for (int i{1}; i < n; i++) {
        cmp = nums[i]; 
        prev = i-1;
        while (prev >= 0 && nums[prev] > cmp) {
            nums[prev+1] = nums[prev];
            prev--;
        }
        nums[prev+1] = cmp;
        printVector(n, nums);
    }
}

int main() {
    int n;
    std::cin >> n;
    int nums[n];
    for (int i{}; i < n; i++)
        std::cin >> nums[i];
    insertionSort2(n, nums);

    return 0;
}